packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,343 @@
+    uint256 resolvedBlock;
+}
+
+/// @title DataAvailabilityChallenge
+/// @notice This contract enables data availability of a data commitment at a given block number to be challenged.
+///         To challenge a commitment, the challenger must first post a bond (bondSize).
+///         Challenging a commitment is only possible within a certain block interval (challengeWindow) after the
+///         commitment was made.
+///         If the challenge is not resolved within a certain block interval (resolveWindow), the challenge can be
+///         expired.
+///         If a challenge is expired, the challenger's bond is unlocked and the challenged commitment is added to the
+///         chain of expired challenges.
+contract DataAvailabilityChallenge is OwnableUpgradeable, ISemver {
+    /// @notice Error for when the provided resolver refund percentage exceeds 100%.
+    error InvalidResolverRefundPercentage(uint256 invalidResolverRefundPercentage);
+
+    /// @notice Error for when the challenger's bond is too low.
+    error BondTooLow(uint256 balance, uint256 required);
+
+    /// @notice Error for when attempting to challenge a commitment that already has a challenge.
+    error ChallengeExists();
+
+    /// @notice Error for when attempting to resolve a challenge that is not active.
+    error ChallengeNotActive();
+
+    /// @notice Error for when attempting to unlock a bond from a challenge that is not expired.
+    error ChallengeNotExpired();
+
+    /// @notice Error for when attempting to challenge a commitment that is not in the challenge window.
+    error ChallengeWindowNotOpen();
+
+    /// @notice Error for when the provided input data doesn't match the commitment.
+    error InvalidInputData(bytes32 providedDataHash, bytes32 expectedHash);
+
+    /// @notice Error for when the call to withdraw a bond failed.
+    error WithdrawalFailed();
+
+    /// @notice An event that is emitted when the status of a challenge changes.
+    /// @param challengedHash The hash of the commitment that is being challenged.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param status The new status of the challenge.
+    event ChallengeStatusChanged(
+        bytes32 indexed challengedHash, uint256 indexed challengedBlockNumber, ChallengeStatus status
+    );
+
+    /// @notice An event that is emitted when the bond size required to initiate a challenge changes.
+    event RequiredBondSizeChanged(uint256 challengeWindow);
+
+    /// @notice An event that is emitted when the percentage of the resolving cost to be refunded to the resolver
+    /// changes.
+    event ResolverRefundPercentageChanged(uint256 resolverRefundPercentage);
+
+    /// @notice An event that is emitted when a user's bond balance changes.
+    event BalanceChanged(address account, uint256 balance);
+
+    /// @notice Semantic version.
+    /// @custom:semver 0.0.0
+    string public constant version = "0.0.0";
```

Comment:
We will need to be sure to update this to `1.0.0` for any prod deployments, so both long lived testnet + mainnet